File: Jim/leetcode/11.27/ContainerWithMostWater.js
/**
 * 往小的那邊前進
 * 面積 (right - left) * Math.min(height[right], height[left])
 * 用current紀錄當次 再和an
 * @param {number[]} height
 * @return {number}
 */
let maxArea = (height) => {
    let ans = 0;
    let length = height.length;
    let left = 0;
    let right = length - 1;
    if (length <= 1) return height[0];

    for (let i = 0; i < length; i++) {
        let current = (right - left) * Math.min(height[right], height[left]);
        ans = Math.max(ans, current);
        (height[left] <= height[right]) ? left++ : right--;
    }
    return ans;
};

File: Henry/Week 2/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int Left = 0;
        int Right = height.size() - 1;
        int Max = 0;
        int Ans = 0;
        for(int i = 0; i < height.size();i++){
            if(height[Left]>height[Right]){
                Ans = height[Right]*(Right-Left);
                Right--;
            }
            else{
                 Ans = height[Left]*(Right-Left);
                 Left++;
            }
            if(Ans>Max)Max = Ans;
        }
        return Max;
    }
};
